Mathlib/Algebra/GroupWithZero/NonZeroDivisors.lean

```diff
@@ -150,6 +150,9 @@ theorem mul_right_coe_nonZeroDivisors_eq_zero_iff {c : M₀⁰} : x * c = 0 ↔
 lemma IsUnit.mem_nonZeroDivisors (hx : IsUnit x) : x ∈ M₀⁰ :=
   ⟨fun _ ↦ hx.mul_right_eq_zero.mp, fun _ ↦ hx.mul_left_eq_zero.mp⟩
 
+variable (M₀) in
+lemma isUnit_le_nonZeroDivisors : IsUnit.submonoid M₀ ≤ M₀⁰ := fun _ ↦ (·.mem_nonZeroDivisors)
```

Comment:
Do we really need this? It's *very close* to `mem_nonZeroDivisors`.

Reply:
This is used in a rewriting step in `self_iff_nonZeroDivisors_le_isUnit`, and the proof doesn't work with `and_iff_left fun _ ↦ (·.mem_nonZeroDivisors)` (needs `and_iff_left (by exact fun _ ↦ (·.mem_nonZeroDivisors))`), so I prefer to have this lemma.